kie-api/src/main/java/org/kie/api/io/ResourceType.java

```diff
@@ -248,12 +268,8 @@ public static ResourceType getResourceType(final String resourceType) {
     }
 
     public static ResourceType determineResourceType(final String resourceName) {
-        for (ResourceType type : CACHE.values()) {
-            if (type.matchesExtension(resourceName)) {
-                return type;
-            }
-        }
-        return null;
+        int dotPos = resourceName.lastIndexOf( '.' );
+        return dotPos < 0 ? null : CACHE.get( resourceName.substring( dotPos ) );
```

Comment:
This change can introduce a bug if (or when) we have a resource type extension that contains `.` (i.e. `gwt.xml`). I understand a `get` on a map is much more efficient than the previous version but I think it is a premature optimization.

Reply:
Functionally the public API of this class behaves exactly as before, I made just a perf improvement. If this breaks something (I don't think so) I'm hoping we will realized this with tests.

Reply:
It doesn't break now because we have no `ResourceType` that has a `.` as part of the extension.
This code assumes a constraint on extension format (aka split on ``lastIndexOf `.`)`` to get extension) that is implicit and not enforced.
The method used in the previous code ([matchesExtension](https://github.com/kiegroup/droolsjbpm-knowledge/blob/bc2f3aae35e4d7a419abe8afa2751cd931587118/kie-api/src/main/java/org/kie/api/io/ResourceType.java#L278)) doesn't suffer of the same issue because the used condition is `endsWith("." + defaultExtension)`

In short, this type is legit and breaks the proposed code
```java
public static final ResourceType BROKEN = addResourceTypeToRegistry("BROKEN",
                                                                    "Broken resrouce type",
                                                                    "src/main/resources",
                                                                    "broken.extension");
```